src/shazamkit.cs

```diff
@@ -0,0 +1,250 @@
+using System;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+using AVFoundation;
+
+namespace ShazamKit {
+
+	[Native]
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[ErrorDomain ("SHErrorDomain")]
+	public enum SHErrorCode : long
+	{
+		InvalidAudioFormat = 100,
+		AudioDiscontinuity = 101,
+		SignatureInvalid = 200,
+		SignatureDurationInvalid = 201,
+		MatchAttemptFailed = 202,
+		CustomCatalogInvalid = 300,
+		CustomCatalogInvalidURL = 301,
+		MediaLibrarySyncFailed = 400,
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[Static]
+	enum SHMediaItemProperty
+	{
+		[Field ("SHMediaItemShazamID")]
+		ShazamID,
+		[Field ("SHMediaItemTitle")]
+		Title,
+		[Field ("SHMediaItemSubtitle")]
+		Subtitle,
+		[Field ("SHMediaItemArtist")]
+		Artist,
+		[Field ("SHMediaItemWebURL")]
+		WebURL,
+		[Field ("SHMediaItemAppleMusicID")]
+		AppleMusicID,
+		[Field ("SHMediaItemAppleMusicURL")]
+		AppleMusicURL,
+		[Field ("SHMediaItemArtworkURL")]
+		ArtworkURL,
+		[Field ("SHMediaItemVideoURL")]
+		VideoURL,
+		[Field ("SHMediaItemExplicitContent")]
+		ExplicitContent,
+		[Field ("SHMediaItemGenres")]
+		Genres,
+		[Field ("SHMediaItemISRC")]
+		Isrc,
+		[Field ("SHMediaItemMatchOffset")]
+		MatchOffset,
+		[Field ("SHMediaItemFrequencySkew")]
+		FrequencySkew,
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHCatalog
+	{
+		[Export ("minimumQuerySignatureDuration")]
+		double MinimumQuerySignatureDuration { get; }
+
+		[Export ("maximumQuerySignatureDuration")]
+		double MaximumQuerySignatureDuration { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (SHCatalog))]
+	interface SHCustomCatalog
+	{
+		[Export ("addReferenceSignature:representingMediaItems:error:")]
+		bool Add (SHSignature signature, SHMediaItem[] mediaItems, [NullAllowed] out NSError error);
+
+		[Export ("addCustomCatalogFromURL:error:")]
+		bool Add (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("writeToURL:error:")]
+		bool WriteL (NSUrl url, [NullAllowed] out NSError error);
+
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
+		[Export ("new")]
+		[return: Release]
+		SHCustomCatalog New ();
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMatch : NSSecureCoding
+	{
+		[Export ("mediaItems", ArgumentSemantic.Strong)]
+		SHMatchedMediaItem[] MediaItems { get; }
+
+		[Export ("querySignature", ArgumentSemantic.Strong)]
+		SHSignature QuerySignature { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (SHMediaItem))]
```

Comment:
there's no way to set the values for the properties, that's usually because the instance are created by the OS (and not the user code). 
IOW `	[DisableDefaultCtor]`